ethers-signers/src/client.rs

```diff
@@ -74,8 +74,35 @@ pub struct Client<P, S> {
     pub(crate) signer: Option<S>,
     pub(crate) address: Address,
     pub(crate) gas_oracle: Option<Box<dyn GasOracle>>,
+    pub(crate) nonce_manager: RwLock<NonceManager>,
 }
 
+use tokio::sync::RwLock;
+
+#[derive(Clone, Debug)]
+pub(crate) struct NonceManager {
+    initialized: bool,
+    nonce: U256,
```

Comment:
use `AtomicU64` internally and convert to a U256 for the pub api? Unless you're anticipating needing nonces above `u64::MAX`

Reply:
Nonce may be limited to 2^63 - 1 soon, so atomic is enough